File: Glucloser/app/src/main/java/com/nlefler/glucloser/models/BloodSugarParcelable.java
package com.nlefler.glucloser.models;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.Date;

/**
 * Created by Nathan Lefler on 1/4/15.
 */
public class BloodSugarParcelable implements Parcelable {
    private String id;
    private int value;
    private Date date;

    public BloodSugarParcelable() {

    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public int getValue() {
        return value;
    }

    public void setValue(int value) {
        this.value = value;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    /** Parcelable */
    protected BloodSugarParcelable(Parcel in) {
        id = in.readString();
        value = in.readInt();
        long time = in.readLong();
        if (time > 0) {
            date = new Date(time);
        }
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(id);
        dest.writeInt(value);
        if (date != null) {
            dest.writeLong(date.getTime());
        }
    }

    @SuppressWarnings("unused")
    public static final Parcelable.Creator<BloodSugarParcelable> CREATOR = new Parcelable.Creator<BloodSugarParcelable>() {
        @Override
        public BloodSugarParcelable createFromParcel(Parcel in) {
            return new BloodSugarParcelable(in);
        }

        @Override
        public BloodSugarParcelable[] newArray(int size) {
            return new BloodSugarParcelable[size];
        }
    };
}

File: Glucloser/app/src/main/java/com/nlefler/glucloser/models/BloodSugarParcelable.kt
package com.nlefler.glucloser.models

import android.os.Parcel
import android.os.Parcelable

import java.util.Date

/**
 * Created by Nathan Lefler on 1/4/15.
 */
public data class BloodSugarParcelable() : Parcelable {
    public var id: String? = null
    public var value: Int = 0
    public var date: Date? = null

    /** Parcelable  */
    protected constructor(parcel: Parcel) : this() {
        id = parcel.readString()
        value = parcel.readInt()
        val time = parcel.readLong()
        if (time > 0) {
            date = Date(time)
        }
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(id)
        dest.writeInt(value)
        if (date != null) {
            dest.writeLong(date!!.time)
        }
    }

    companion object {
        public val CREATOR: Parcelable.Creator<BloodSugarParcelable> = object : Parcelable.Creator<BloodSugarParcelable> {
            override fun createFromParcel(`in`: Parcel): BloodSugarParcelable {
                return BloodSugarParcelable(`in`)
            }

            override fun newArray(size: Int): Array<BloodSugarParcelable> {
                return Array(size, {i -> BloodSugarParcelable()})
            }
        }
    }
}
